src/psyclone/domain/lfric/kernel/lfric_kernel_metadata.py

```diff
@@ -0,0 +1,506 @@
+            spec_part, "operates_on").string
+        kernel_metadata.operates_on = value
+
+        # the value of index offset (NE, ...)
+        try:
+            value = LFRicKernelMetadata._get_property(
+                spec_part, "gh_shape").string
+            kernel_metadata.gh_shape = value
+        except ParseError:
+            kernel_metadata.gh_shape = None
+
+        # the name of the procedure that this metadata refers to.
+        kernel_metadata.procedure_name = LFRicKernelMetadata._get_property(
+            spec_part, "code").string
+
+        # meta_args contains arguments which have
+        # properties. Therefore create appropriate (ScalarArg,
+        # FieldArg, ...) instances to capture this information.
+        kernel_metadata._meta_args = LFRicKernelMetadata._get_property(
+            spec_part, "meta_args")
+        args = walk(kernel_metadata.meta_args, Fortran2003.Ac_Value_List)
+        if not args:
+            raise ParseError(
+                f"meta_args should be a list, but found "
+                f"'{str(kernel_metadata.meta_args)}' in '{spec_part}'.")
+
+        kernel_metadata._meta_args = []
+        for meta_arg in args[0].children:
+            form = meta_arg.children[1].children[0].tostr()
+            form = form.lower()
+            if form == "gh_scalar":
+                arg = ScalarArg.create_from_psyir(meta_arg)
+            elif form == "gh_operator":
+                arg = OperatorArg.create_from_psyir(meta_arg)
+            elif form == "gh_columnwise_operator":
+                arg = ColumnwiseOperatorArg.create_from_psyir(meta_arg)
+            elif "gh_field" in form:
+                vector_arg = "gh_field" in form and "*" in form
+                nargs = len(meta_arg.children[1].children)
+                intergrid_arg = False
+                if nargs == 5:
+                    fifth_arg = meta_arg.children[1].children[4]
+                    if fifth_arg.children[0].string == "mesh_arg":
+                        intergrid_arg = True
+
+                if intergrid_arg and vector_arg:
+                    arg = InterGridVectorArg.create_from_psyir(meta_arg)
+                elif intergrid_arg and not vector_arg:
+                    arg = InterGridArg.create_from_psyir(meta_arg)
+                elif vector_arg and not intergrid_arg:
+                    arg = FieldVectorArg.create_from_psyir(meta_arg)
+                else:
+                    arg = FieldArg.create_from_psyir(meta_arg)
+            else:
+                raise ParseError(
+                    f"Expected a 'meta_arg' entry to be a "
+                    f"field, a scalar or an operator, but found "
+                    f"'{meta_arg}'.")
+            kernel_metadata.meta_args.append(arg)
+
+        try:
+            meta_funcs = LFRicKernelMetadata._get_property(
+                spec_part, "meta_funcs")
+            args = walk(meta_funcs, Fortran2003.Ac_Value_List)
```

Comment:
Please add a TODO. Along with that I think it will probably be better to have separate (internal) methods to handle meta_args, meta_funcs, meta_reference_element, etc...

Reply:
Added TODO's

Reply:
already done.